### PR TITLE
Receiver parameter & ETH zap contract

### PR DESCRIPTION
**Pull request by sydneyitguy**
Another last-minute change involves adding a receiver parameter to the mint() and burn() functions. This will enhance flexibility, allowing the bond to work with external contracts such as Zap with less gas fees.

Community audit participants, please review the changes 🙏🏻

### CODE REVIEW

File: contracts/MCV2_ZapV1.sol [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        (,,,,reserveToken,) = BOND.tokenBond(token);
+    }
+
+    /**
+     * @dev Mint tokens by sending ETH.
+     * @param token The token address.
+     * @param tokensToMint The amount of tokens to mint.
+     * @param receiver The address to receive the minted tokens.
+     */
+    function mintWithEth(address token, uint256 tokensToMint, address receiver) external payable {
```
Comment (0x3agle) [inline contracts/MCV2_ZapV1.sol:53]: Add a check to ensure that `receiver != address(0)`

**Comment by 0x3agle** [comment]
Considering that it can receive ETH from anyone, I suggest adding a `rescueETH(amount, receiver) onlyOwner` function to prevent ETH from getting stuck in this contract.

File: contracts/MCV2_ZapV1.sol (hunk 1 above)
Comment (0x3agle) [reply contracts/MCV2_ZapV1.sol:54]: @sydneyitguy I noticed that this parameter is also added in the Bond contract's mint and burn functions. So, I recommend adding this validation in those functions too.

Comment (sydneyitguy) [reply contracts/MCV2_ZapV1.sol:54]: Oh you're right! 
Thanks for pointing that out :)

**Pull request merged**